crates/neon/src/types_impl/buffer/types.rs

```diff
@@ -0,0 +1,783 @@
+use std::{marker::PhantomData, slice};
+
+use crate::{
+    context::{internal::Env, Context},
+    handle::{internal::TransparentNoCopyWrapper, Handle, Managed},
+    object::Object,
+    result::{JsResult, Throw},
+    sys::{self, raw, TypedArrayType},
+    types::buffer::{
+        lock::{Ledger, Lock},
+        private::{self, JsTypedArrayInner},
```

Comment:
This is confusing because of all the funky re-exporting at the top level `lib.rs` -- `neon::types` actually just _is_ `neon::types_impl`. But I'll change these `use` declarations to import more directly from `types_impl` so it's more self-explanatory.